src/main/java/life/qbic/model/download/OpenbisConnector.java

```diff
@@ -77,6 +76,30 @@ public DataSetPermId registerDataset(Path uploadPath, String experimentID,
     return null;
   }
 
+  public DataSetPermId registerDatasetForSample(Path uploadPath, String sampleID,
+      List<String> parentCodes) {
+    UploadedDataSetCreation creation = prepareDataSetCreation(uploadPath, parentCodes);
+    creation.setSampleId(new SampleIdentifier(sampleID));
+
+    try {
+      return openBIS.createUploadedDataSet(creation);
+    } catch (final Exception e) {
+      LOG.error(e.getMessage());
+    }
+    return null;
```

Comment:
Why does this method specify a return value if it always returns a null value?

Reply:
it returns a permID if the try case succeeds

Reply:
I'm blind thanks :D